sympy/sets/fancysets.py

```diff
@@ -472,14 +478,17 @@ def __new__(cls, *args):
         if start.is_infinite:
             end = stop
         else:
-            ref = start if start.is_finite else stop
-            n = ceiling((stop - ref)/step)
-            if n <= 0:
-                # null Range
-                start = end = S.Zero
-                step = S.One
+            if not all(w.is_number for w in [start, stop, step]):
+                end = stop
             else:
-                end = ref + n*step
+                ref = start if start.is_finite else stop
+                n = ceiling((stop - ref)/step)
+                if n <= 0:
```

Comment:
You are modifying this function to allow symbolic objects, right? I would use `(n <= 0) == True` as the operator `<=` may **not** return a boolean with symbols.

Reply:
I thought this initially, too. This is the part of the code where we have confirmed that all values are numbers.